pkg/metrics/azure_metrics.go

```diff
@@ -99,6 +99,9 @@ func (mc *MetricContext) ObserveOperationWithResult(isOperationSucceeded bool, l
 	resultCode := "succeeded"
 	if !isOperationSucceeded {
 		resultCode = "failed"
+		if len(mc.attributes) > 0 {
```

Comment:
UT test needs 1.2.0 logr, so UT test is temporarily removed and will be added in the future.